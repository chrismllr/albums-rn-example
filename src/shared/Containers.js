// @flow
import glamorous from 'glamorous-native';
import { palette, spacing } from '../styles/common';

export const Container = glamorous.view({
  flex: 1,
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: palette.white,
  padding: spacing.sp4
});

type FlexProps = {
  vSpacing: number,
  vOffset: number,
  hSpacing: number,
  hOffset: number,

  flexDirection: string,
  alignItems: string,
  justifyContent: string,

  flex: number
};

export const FlexContainer = glamorous.view({
}, (props: FlexProps) => ({
  paddingTop: props.vSpacing,
  paddingBottom: props.vSpacing,
  marginTop: props.vOffset,
  marginBottom: props.vOffset,

  paddingLeft: props.hSpacing,
  paddingRight: props.hSpacing,
  marginLeft: props.hOffset,
  marginRight: props.hOffset,

  flexDirection: props.flexDirection,
  alignItems: props.alignItems,
  justifyContent: props.justifyContent,
  flex: props.flex
}));
