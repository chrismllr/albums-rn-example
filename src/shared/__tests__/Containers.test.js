import React from 'react';
import 'react-native';
import { Container } from '../Containers';
import renderer from 'react-test-renderer';

describe('Container', () => {
  it('renders to snapshot', () => {
    const tree = renderer.create(
      <Container />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
