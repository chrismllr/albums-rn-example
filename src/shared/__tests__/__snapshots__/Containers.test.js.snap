// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders to snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "center",
        "padding": 12,
      },
    ]
  }
/>
`;
